src/marqo/core/models/marqo_index.py

```diff
@@ -270,6 +291,34 @@ def _cache_or_get(self, key: str, func):
         if key not in self._cache:
             self._cache[key] = func()
         return self._cache[key]
+    
+    def get_text_query_prefix(self, request_level_prefix: Optional[str] = None) -> str:
```

Comment:
Moved them to Model.

Also, regarding why i put request_level_prefix here instead of just directly using them in tensor_search is for easier code readability. I did not want to have to repeat the conditional to check whether the request level prefix is none or a string, so that logic was moved here.